### ISSUE TITLE
Bug: Event handlers are no longer bound in edge case

### ISSUE THREAD

**Issue by mindeavor** [marko-js/marko#653]
## Bug Report

After two renders, a click handler no longer triggers. This bug was introduced in 4.2.0 (4.2.3 still has the bug, but 4.1.3 works perfectly).

### Context

```marko
class {
  onCreate(input) {
    this.state = { mode: input.mode || 'sign-up' }
  }
  setType(value) {
    this.state.mode = value
  }
}

<form class="session-panel">

  <div class="sign-up" if(state.mode === 'sign-up')>
    <h3>Create Account</h3>
    <div class="switch">
      Already have an account?
      <sign-in-link />
    </div>
  </div>

  <div class="sign-in" if(state.mode === 'sign-in')>
    <h3>Sign In</h3>

    <div class="switch">
      Don't have an account?
      <sign-up-link />
    </div>
  </div>
</form>

<macro sign-in-link>
  <a on-click('setType', 'sign-in')>&nbsp;Sign&nbsp;in&nbsp;›</a>
</macro>

<macro sign-up-link>
  <a on-click('setType', 'sign-up')>&nbsp;Create&nbsp;your&nbsp;account&nbsp;›</a>
</macro>
```

1. Click "Sign in"
2. Click "Create your account"
3. Click "Sign in" again

### Expected Behavior

It should go back to the Sign in mode.

### Actual Behavior

Nothing happens :)

<details><summary>Additional Info</summary>

### Your Environment

* Version used: 4.2.x
* Environment name and version: Chrome 56
* Operating System and version (desktop or mobile): OS X 10.11
</details>

**Comment by patrick-steele-idem** [comment]
Found the problem. Working on a fix.

**Comment by patrick-steele-idem** [comment]
New version published with fix: `marko@4.2.4`

Thanks for reporting the problem @mindeavor and making it easy to reproduce as always!
